packages/mask/src/plugins/Profile/SNSAdaptor/hooks/useFootprints.ts

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from 'react'
+import type { GeneralAssetWithTags } from '../common/types'
+import utils from '../common/utils'
+
+export function useFootprints() {
+    const [footprints, setFootprints] = useState<GeneralAssetWithTags[]>([])
+    const [isLoading, setIsLoading] = useState(true)
+
+    useEffect(() => {
```

Comment:
Refactor to use [`useAsync`](https://github.com/streamich/react-use/blob/master/docs/useAsync.md).